pallets/parachain-staking/src/set.rs

```diff
@@ -87,3 +88,84 @@ impl<T: Ord> From<Vec<T>> for OrderedSet<T> {
         Self::from(v)
     }
 }
+
+/// An ordered set backed by `BoundedVec`
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, TypeInfo, Clone, MaxEncodedLen)]
+#[scale_info(skip_type_params(S))]
+pub struct BoundedOrderedSet<T, S: Get<u32>>(pub BoundedVec<T, S>);
+
+impl<T, S: Get<u32>> sp_std::default::Default for BoundedOrderedSet<T, S> {
+    fn default() -> Self {
+        BoundedOrderedSet(BoundedVec::default())
+    }
+}
+
+impl<T: Ord, S: Get<u32>> BoundedOrderedSet<T, S> {
+    /// Create a new empty set
+    pub fn new() -> Self {
+        Self(BoundedVec::default())
+    }
+
+    /// Create a set from a `Vec`.
+    /// `v` will be sorted and dedup first.
+    pub fn from(mut v: BoundedVec<T, S>) -> Self {
+        v.sort();
+        let v = v.try_mutate(|inner| inner.dedup()).expect(
+            "input is a valid BoundedVec and deduping can only reduce the number of entires; qed",
+        );
+        Self::from_sorted_set(v)
+    }
+
+    /// Create a set from a `Vec`.
+    /// Assume `v` is sorted and contain unique elements.
+    pub fn from_sorted_set(v: BoundedVec<T, S>) -> Self {
```

Comment:
I know this mimics the interface of the OrderedSet, but why can't we use from instead? Is it for speed?
This function doesn't "honor" its name, it uses a bounded vector not a set as an input.
It feels prone to error especially as it depends on the input to be unique.

Reply:
we can use `from` just a remnant of an experiment I did , it will be removed